player/src/webrtc/Signaling.ts

```diff
@@ -0,0 +1,227 @@
+import debug from 'debug'
+
+import {
+  Authenticator,
+  Environment,
+  IceSignalingMessage,
+  InitSessionMessage,
+  Message,
+  SDPSignalingMessage,
+  WebRTCVideoReceive,
+} from './types'
+
+const debugLog = debug('msp:webrtc-video')
+
+export interface SignalingListener {
+  onIceServers: (iceServers: Array<RTCIceServer>) => void
+  onRemoteIceCandidate: (
+    iceCandidate: RTCIceCandidateInit
+  ) => void
+  onRemoteSessionDescription: (
+    signaling: Signaling,
+    sessionDescription: RTCSessionDescriptionInit
+  ) => void
+  onError: (
+    errorCode: string,
+    errorDescription: string
+  ) => void
+}
+
+export class Signaling {
+  sessionId = ''
+  authenticator: Authenticator
+  organizationArn: string
+  targetId: string
+  env: Environment
+  options: WebRTCVideoReceive
+  listener: SignalingListener
+  ws?: WebSocket
+
+  constructor(
+    authenticator: Authenticator,
+    organizationArn: string,
+    targetId: string,
+    options: WebRTCVideoReceive,
+    listener: SignalingListener,
+    env: Environment = 'prod'
+  ) {
+    this.sessionId = crypto.randomUUID()
+    this.authenticator = authenticator
+    this.organizationArn = organizationArn
+    this.targetId = targetId
+    this.env = env
+    this.options = options
+    this.listener = listener
+  }
+
+  async connect() {
+    if (this.ws !== undefined) {
+      this.ws.close(1000, 're-initializing connection')
+    }
+
+    const baseUrl = `wss://signaling.${this.env}.webrtc.connect.axis.com/client`
```

Comment:
So this is hardcoded to Axis "internal" webrtc client?

Why not allow someone to use their own and let them specify a baseUrl?

I'm especially not fond of passing around environments and let that be the only thing someone can specify.